bilby_numpyro/numpyro.py

```diff
@@ -146,6 +170,53 @@ def write_current_state(self):
         """
         pass
 
+    def _run_with_checkpointing(self, mcmc, sample_key):
+
+        # first run
+        if not os.path.isfile(self.outdir + "/numpyro_checkpt.pkl"):
+            
+            mcmc.run(sample_key)
+
+            checkpt = {}
+            checkpt['samples'] = mcmc.get_samples()
+            checkpt['state'] = mcmc.last_state
+            checkpt['num_samples'] = (checkpt['samples'][list(checkpt['samples'].keys())[0]]).size
+            checkpt['log_likelihood'] = log_likelihood(mcmc.sampler.model, mcmc.get_samples())
+
+            logger.info('checkpointing at ' + str(checkpt['num_samples']) + ' samples')
+            with open(self.outdir + "/numpyro_checkpt.pkl", 'wb') as f:
+                pickle.dump(checkpt, f)
+
+        else:
+            logger.info('loading from checkpoint')
+            with open(self.outdir + "/numpyro_checkpt.pkl", 'rb') as f:
+                checkpt = pickle.load(f)
+
+        while checkpt['num_samples'] < self.kwargs['num_total_samples']:
+
+            # fix checkpoint state as post warmup state
+            mcmc.post_warmup_state = checkpt['state'] 
+            mcmc.run(sample_key)
```

Comment:
Good point. I assumed that it is ok since the state of the sampler changes at each point. But just to be safe, now I split the key every time it checkpoints and run with the new key so that it still is reproducible.